### PR TITLE
Standardizing fxa utm parameters and values (#7061)

### PR DESCRIPTION
**Pull request by amychurchwell**
## Description
Standardizing FxA parameters so they are uniform across the site.

## Issue / Bugzilla link
#7061

## Testing
- Format for `entrypoint`: `domain-point_of_CTA`

### CODE REVIEW

File: bedrock/firefox/templates/firefox/accounts-2018.html [hunk 1]
```diff
@@ -289,7 +290,13 @@ <h2>{{ _('You deserve peace of mind everywhere.') }}</h2>
         </div>
 
         <div class="show-fxa-supported-signed-out">
-          {{ fxa_email_form(entrypoint='accounts-page', utm_source='accounts-page', utm_params={'campaign': 'fxa-embedded-form', 'content': 'accounts-page', 'medium': 'referral'}, button_class='mzp-c-button mzp-t-primary mzp-t-product', button_text=_('Create an Account')) }}
+          {{ fxa_email_form(
+            entrypoint='mozilla.org-fxa_benefits',
+            utm_source='mozilla.org-fxa_benefits',
+            utm_params={'referrer': 'https://www.mozilla.org/en-US/firefox/accounts/'},
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/accounts-2018.html]: Is there a need to add `referrer` here? From our convo yesterday, it sounded like this is only necessary if `entrypoint` is missing. If we do require adding this, we should make sure it's adding the correct locale in the URL.

File: bedrock/firefox/templates/firefox/accounts-2018.html [hunk 2]
```diff
@@ -38,7 +38,8 @@ <h1 class="mzp-c-hero-title">{{ _('One log-in. Power and privacy everywhere.') }
       {% endif %}
       <div id="hero-cta" class="mzp-c-hero-cta">
         <div class="show-fxa-supported-signed-out">
-          <a {{ fxa_link_fragment(entrypoint='accounts-page', action='signup', utm_params={'campaign': 'fxa-benefits-page', 'content': 'accounts-page-top-cta', 'medium': 'referral', 'source': 'accounts-page'}) }} class="mzp-c-button mzp-t-primary mzp-t-product" data-button-name="Create account" data-link-type="button" data-cta-position="primary cta" id="features-hero-account">
+          <a
+          {{ fxa_link_fragment(entrypoint='mozilla.org-accounts-page', action='signup', utm_params={'campaign': 'fxa-benefits-page', 'content': 'accounts-page-top-cta', 'medium': 'referral', 'source': 'accounts-page'}) }} class="mzp-c-button mzp-t-primary mzp-t-product" data-cta-position="primary cta" id="features-hero-account">
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/accounts-2018.html]: Reading the updated issue [description](https://github.com/mozilla/bedrock/issues/7061#issue-431606721), it sounds like this should be `mozilla.org-accounts_page`? (using underscores to separate the point of CTA)

File: bedrock/firefox/templates/firefox/accounts-2018.html [hunk 3]
```diff
@@ -289,7 +290,13 @@ <h2>{{ _('You deserve peace of mind everywhere.') }}</h2>
         </div>
 
         <div class="show-fxa-supported-signed-out">
-          {{ fxa_email_form(entrypoint='accounts-page', utm_source='accounts-page', utm_params={'campaign': 'fxa-embedded-form', 'content': 'accounts-page', 'medium': 'referral'}, button_class='mzp-c-button mzp-t-primary mzp-t-product', button_text=_('Create an Account')) }}
+          {{ fxa_email_form(
+            entrypoint='mozilla.org-fxa_benefits',
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/accounts-2018.html]: Is there a reason that this `entrypoint` is different to the one above, if it's coming from the same page?

File: bedrock/firefox/templates/firefox/accounts.html [hunk 4]
```diff
@@ -38,7 +38,7 @@ <h1>{{_('Sign up, sign in, stay in sync.')}}</h1>
       <div class="fxa-cta show-fxa-supported-signed-in show-fxa-supported-signed-out">
         <p class="show-fxa-supported-signed-in signed-in">{{_('You’re signed in and ready to start using Sync.')}}</p>
         <div class="show-fxa-supported-signed-out">
-          {{ fxa_email_form(entrypoint='accounts-page', utm_source='accounts-page', utm_params={'campaign': 'fxa-embedded-form', 'content': 'accounts-page', 'medium': 'referral'}) }}
+          {{ fxa_email_form(entrypoint='mozilla.org-accounts', utm_source='accounts-page', utm_params={'campaign': 'fxa-embedded-form', 'content': 'accounts-page', 'medium': 'referral'}) }}
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/accounts.html]: Should this be `mozilla.org-accounts_page` to match the other template, or is it better being something unique?

File: bedrock/firefox/templates/firefox/new/scene1.html [hunk 5]
```diff
@@ -165,7 +165,7 @@ <h4 class="heading-linux">{{_('Download Firefox <span>for Linux</span>') }}</h4>
   <h4 class="fxa-title">{{ _('You’ve already got the Firefox browser. Now get everything else Firefox.') }}</h4>
   <p class="fxa-intro">{{ _('Sync bookmarks, send tabs, save to Pocket and more with a Firefox Account.') }} <a href="{{ url('firefox.accounts') }}">{{ _('Learn more') }}</a></p>
 
-  {{ fxa_email_form(entrypoint='firefox-new', button_text=_('Get a Firefox Account'), utm_source='firefox-new', utm_params={'campaign': 'firefox-new', 'content': 'firefox-new-fxa', 'term': 'existing-users', 'medium': 'referral'}) }}
+  {{ fxa_email_form(entrypoint='mozilla.org-firefox_new', button_text=_('Get a Firefox Account'), utm_source='firefox-new', utm_params={'campaign': 'firefox-new', 'content': 'firefox-new-fxa', 'term': 'existing-users', 'medium': 'referral'}) }}
 
   <p class="fxa-signin">{{ _('Already have an account?') }} <a {{ fxa_link_fragment(entrypoint='accounts-page', utm_params={'campaign': 'fxa-embedded-form', 'content': '/firefox/new/', 'source': 'download-page'}) }}>{{ _('Sign In') }}</a></p>
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/new/scene1.html]: There's an `entrypoint` here with an incorrect value of `accounts-page`, which seems like it needs updating?

**Comment by alexgibson** [comment]
Looks like there's a couple of entry points that still need updating:

https://github.com/mozilla/bedrock/blob/master/bedrock/firefox/templates/firefox/whatsnew/fx64/whatsnew-fx64-base.html#L72

https://github.com/mozilla/bedrock/blob/master/bedrock/firefox/templates/firefox/firstrun/firstrun-quantum.html#L66-L68 (you may need to rebase on master to get this one that was just added).

File: bedrock/firefox/templates/firefox/accounts.html (hunk 4 above)
Comment (amychurchwell) [reply bedrock/firefox/templates/firefox/accounts.html]: I will update it to match the other template, I'm not sure if they need to be unique.

File: bedrock/firefox/templates/firefox/accounts-2018.html (hunk 1 above)
Comment (amychurchwell) [reply bedrock/firefox/templates/firefox/accounts-2018.html]: really good point/catch. Do we have logic to add `referrer` if `entrypoint` is missing? If not I think that would make a good issue.

Comment (alexgibson) [reply bedrock/firefox/templates/firefox/accounts-2018.html]: I don't know of anywhere where we don't pass `entrypoint` already, so there's no logic for referrer afaik.

**Comment by alexgibson** [comment]
@amychurchwell changes look good, but I think there are a few data-attributes that still need updating (the button in the global nav at least).

**Comment by alexgibson** [comment]
Merging as per: https://github.com/mozilla/bedrock/pull/7113#issuecomment-488332980

**Comment by amychurchwell** [comment]
Updating title of this PR to be more specific about what parameters were touched. UTM work continues in #7397

**Pull request merged**